netbox/project-static/src/util.ts

```diff
@@ -162,10 +163,15 @@ function buildUrl(destination: string): string {
 
   // If the `origin` exists in the API path (as in the case of paginated responses), remove it.
   const origin = new RegExp(window.location.origin, 'g');
-  const path = pathname.replaceAll(origin, '');
+  let path = pathname.replaceAll(origin, '');
 
   const basePath = getBasePath();
 
+  // If the `BASE_PATH` already exists in the URL, remove it.
+  if (basePath !== '' && path.includes(basePath)) {
```

Comment:
I'm not a fan of this approach because IMO it's too vulnerable to false positives. For example, what if a user sets `BASE_PATH = 'ipam/'` (because NetBox is their IPAM app)? That would break IPAM API endpoints.

I'd much prefer to either _always_ prepend the base path, or _never_ prepend it.

Reply:
> I'd much prefer to either always prepend the base path, or never prepend it.

I completely agree with this. I didn't consider there being a naming conflict between netbox paths and the base path, let me take another swing at accounting for that.

Reply:
Was there an issue with the approach I took? It seemed to take a very light touch, and invokes `buildUrl()` only where it's needed.